cmd/spire-server/cli/federation/common.go

```diff
@@ -1,9 +1,160 @@
 package federation
 
 import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/spiffe/go-spiffe/v2/spiffeid"
 	"github.com/spiffe/spire-api-sdk/proto/spire/api/types"
+	"github.com/spiffe/spire/cmd/spire-server/util"
 )
 
+// FederationRelationship type is used for parsing federation relationships from file
+type federationRelationshipJSON struct {
+	TrustDomain           string          `json:"trust_domain,omitempty"`
+	BundleEndpointURL     string          `json:"bundle_endpoint_url,omitempty"`
+	BundleEndpointProfile string          `json:"bundle_endpoint_profile,omitempty"`
+	EndpointSPIFFEID      string          `json:"endpoint_spiffe_id,omitempty"`
+	Bundle                json.RawMessage `json:"bundle,omitempty"`
+	BundleFormat          string          `json:"bundle_format,omitempty"`
+	BundlePath            string
+}
+
+// federationRelationshipsFromFile parse a json file into types FederationRelationships
+func federationRelationshipsFromFile(path string) ([]*types.FederationRelationship, error) {
+	r := os.Stdin
+	if path != "-" {
+		f, err := os.Open(path)
+		if err != nil {
+			return nil, err
+		}
+		defer f.Close()
+		r = f
+	}
+
+	dat, err := io.ReadAll(r)
+	if err != nil {
+		return nil, err
+	}
+
+	relationships := &federationRelationships{}
+	if err := json.Unmarshal(dat, &relationships); err != nil {
```

Comment:
Already a pointer to the struct:
```suggestion
	if err := json.Unmarshal(dat, relationships); err != nil {
```